src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -15,66 +16,88 @@ import scala.collection.mutable
 object ResourceMatcher {
   type Role = String
 
-  case class ScalarMatch(requiredValue: Double, offeredValue: Double, role: String) {
-    def matches: Boolean = requiredValue <= offeredValue
-    def matchingRole: Option[Role] = if (matches) Some(role) else None
+  private[this] val log = LoggerFactory.getLogger(getClass)
+
+  /**
+    * A successful match result of the [[ResourceMatcher]].matchResources method.
+    */
+  case class ResourceMatch(scalarMatches: Iterable[ScalarMatch], portsMatch: PortsMatch) {
+    lazy val hostPorts: Seq[Int] = portsMatch.hostPorts
+
+    def scalarMatch(name: String): Option[ScalarMatch] = scalarMatches.find(_.resourceName == name)
```

Comment:
Only used in tests and I guess you meant it as a test helper – so I'd either make it package private or put it into the test as a helper function